evalml/preprocessing/data_splitters/balanced_classification_splitter.py

```diff
@@ -0,0 +1,91 @@
+
+    def __init__(self, balanced_ratio=4, min_samples=100, min_percentage=0.1, random_seed=0):
+        """
+        Arguments:
+            balanced_ratio (float): The largest majority:minority ratio that is accepted as 'balanced'. For instance, a 4:1 ratio would be
+                represented as 4, while a 6:5 ratio is 1.2. Must be greater than or equal to 1 (or 1:1). Defaults to 4.
+
+            min_samples (int): The minimum number of samples that we must have for any class, pre or post sampling. If a class must be downsampled, it will not be downsampled past this value.
+                To determine severe imbalance, the minority class must occur less often than this and must have a class ratio below min_percentage.
+                Must be greater than 0. Defaults to 100.
+
+            min_percentage (float): The minimum percentage of the minimum class to total dataset that we tolerate, as long as it is above min_samples.
+                If class percentage and min_samples are not met, treat this as severely imbalanced, and we will not resample the data.
+                Must be between 0 and 0.5, inclusive. Defaults to 0.1.
+
+            random_seed (int): The seed to use for random sampling. Defaults to 0.
+        """
+        if balanced_ratio < 1:
+            raise ValueError(f"balanced_ratio must be at least 1, but received {balanced_ratio}")
+        if min_samples <= 0:
+            raise ValueError(f"min_sample must be greater than 0, but received {min_samples}")
+        if min_percentage <= 0 or min_percentage > 0.5:
+            raise ValueError(f"min_percentage must be between 0 and 0.5, inclusive, but received {min_percentage}")
+        self.balanced_ratio = balanced_ratio
+        self.min_samples = min_samples
+        self.min_percentage = min_percentage
+        self.random_seed = random_seed
+        self.random_state = np.random.RandomState(self.random_seed)
+
+    def _find_ideal_samples(self, y):
+        """Returns dictionary of examples to drop for each class if we need to resample
+
+        Arguments:
+            y (pd.Series): Target data passed in
+
+        Returns:
+            (dict): dictionary with undersample target class as key, and number of samples to remove as the value.
+                If we don't need to resample, returns empty dictionary.
+        """
+        counts = y.value_counts()
+        normalized_counts = y.value_counts(normalize=True)
+        class_ratios = normalized_counts.copy().values
+        class_ratios /= class_ratios[-1]
+        # if no class ratios are greater than what we consider balanced, then the target is balanced
+        if all(class_ratios <= self.balanced_ratio):
+            return {}
+        # if any classes have than min_samples counts and are less than min_percentage of the total data,
+        # then it's severely imbalanced
+        if any(counts < self.min_samples) and any(normalized_counts < self.min_percentage):
+            return {}
```

Comment:
Playing devil's advocate here, should severely imbalanced and balanced data produce the same behavior?

The user's data can fall into four cases:
1. Balanced
2. Severely Imbalanced
3. Imbalanced and can downsample
4. Imbalanced but can't downsample because of `min_samples`

Right now users can't tell whether they are in cases 1, 2, or 4 and I think there is value in communicating that. Maybe it belongs in our class imbalance data check though.

Reply:
@freddyaboulton Yeah, I think in discussion with @dsherry, we needed to figure out how to get the class imbalance data check to work with this. Definitely agree that the behavior could be confusing, but I think as long as the data check can help us catch the severely-imbalanced case and warn the user, it clarifies the case.

Reply:
@freddyaboulton just filed a new [issue](https://github.com/alteryx/evalml/issues/1864) to track updating the class imbalance data check